api/worker_grpc.go

```diff
@@ -778,6 +780,70 @@ func notifyWorkersPremieres(daoWrapper dao.DaoWrapper) {
 	}
 }
 
+// FetchLivePreviews gets a live thumbnail from a worker.
+func FetchLivePreviews(daoWrapper dao.DaoWrapper) func() {
+	return func() {
+		workers := daoWrapper.WorkerDao.GetAliveWorkers()
+		liveStreams, err := daoWrapper.StreamsDao.GetCurrentLive(context.Background())
+		if err != nil {
+			return
+		}
+
+		// In case of an error, the preview might be outdated.
+		// That's okay since the cron job is run in 10s again.
+		for _, s := range liveStreams {
+			if s.PlaylistUrl == "" {
+				continue
+			}
+			workerIndex := getWorkerWithLeastWorkload(workers)
+			if len(workers) == 0 {
+				return
+			}
+			conn, err := dialIn(workers[workerIndex])
+			if err != nil {
+				log.WithError(err).Error("Could not connect to worker")
+				endConnection(conn)
+				continue
+			}
+			client := pb.NewToWorkerClient(conn)
+			workers[workerIndex].Workload += 1
+			if err := getLivePreviewFromWorker(&s, workers[workerIndex].WorkerID, client); err != nil {
+				workers[workerIndex].Workload -= 1
+				log.WithError(err).Error("Could not generate live preview")
+				endConnection(conn)
+				continue
+			}
+			workers[workerIndex].Workload -= 1
+		}
+		return
+	}
+}
+
+func getLivePreviewFromWorker(s *model.Stream, workerID string, client pb.ToWorkerClient) error {
+	req := pb.LiveThumbRequest{
+		WorkerID: workerID,
+		HLSUrl:   s.PlaylistUrl,
+	}
+	resp, err := client.GenerateLiveThumbs(context.Background(), &req)
+	if err != nil {
+		log.WithError(err).Error("Can't retrieve raw live preview data from worker")
```

Comment:
return `fmt.Errorf("something: %w", err)` and only log further up the call stack.